src/Behat/Behat/Output/Node/Printer/JUnit/JUnitSuitePrinter.php

```diff
@@ -44,7 +44,7 @@ public function printHeader(Formatter $formatter, Suite $suite)
 
         /** @var JUnitOutputPrinter $outputPrinter */
         $outputPrinter = $formatter->getOutputPrinter();
-        $outputPrinter->createNewFile($suite->getName());
+        $outputPrinter->createNewFile(!empty($formatter->fileName) ? $formatter->fileName : $suite->getName());
```

Comment:
Here. Use delegation instead of control (also known as "Tell, don't ask" principle).

Can you refactor this piece of code to avoid employing creating and reading public properties from the formatter?